packages/pluggableWidgets/accordion-web/src/components/Accordion.tsx

```diff
@@ -0,0 +1,40 @@
+import { createContext, createElement, Dispatch, ReactElement, useMemo, useReducer, useRef } from "react";
+
+import AccordionGroup, { AccGroup } from "./AccordionGroup";
+import { AccordionGroupsReducerAction, getAccordionGroupsReducer } from "../utils/AccordionGroupStateReducer";
+import { AccordionContainerProps } from "../../typings/AccordionProps";
+
+export interface AccordionProps extends Pick<AccordionContainerProps, "class" | "style" | "tabIndex"> {
+    id: string;
+    groups: AccGroup[];
+    collapsible: boolean;
+    singleExpandedGroup?: boolean;
+}
+
+export const AccordionGroupsDispatch = createContext<Dispatch<AccordionGroupsReducerAction> | undefined>(undefined);
+
+export default function Accordion(props: AccordionProps): ReactElement | null {
+    const previousGroupsPropValue = useRef(props.groups);
+
+    const [accordionGroups, accordionGroupsDispatch] = useReducer(
+        getAccordionGroupsReducer(props.collapsible ? (props.singleExpandedGroup ? "single" : "multiple") : "all"), // the accordion group reducer function doesn't need to change during the lifetime of this component, since the singleExpandedGroup won't change.
+        props.groups.map(group => (props.collapsible ? { ...group, collapsed: true } : { ...group, collapsed: false }))
+    );
+
+    if (props.groups !== previousGroupsPropValue.current) {
```

Comment:
Question: wouldn't it better to make this a check onto the content of the arrays rather than rely on the `useMemo` from the parent component? Right now this will trigger anyways if let say an update happens with the same content through dynamic values. In that sense I feel like doing a content comparison is more reilable

Reply:
As discussed we created a separate story on the backlog for this.